cloudinit/sources/DataSourceEc2.py

```diff
@@ -862,37 +869,23 @@ def _collect_platform_data():
     On Ec2 instances experimentation is that product_serial is upper case,
     and product_uuid is lower case.  This returns lower case values for both.
     """
-    data = {}
-    try:
+    uuid = None
+    with suppress(OSError):
```

Comment:
> Did we check that no other exceptions than OSError cannot happen while executing util.load_text_file? If unsure, we could do the following to maintain old behavior:

Good question. I think that this is pretty safe. I suppose this could possibly throw `UnicodeDecodeError` if a malformed UUID exists which cannot be converted to utf-8, so I'll add that exception handler, but I think that all other cases should be handled.

Let's take a look at the relevant functions:

```python3
def pipe_in_out(in_fh, out_fh, chunk_size=1024, chunk_cb=None):
    bytes_piped = 0
    while True:
        data = in_fh.read(chunk_size)
        if len(data) == 0:
            break
        else:
            out_fh.write(data)
            bytes_piped += len(data)
            if chunk_cb:
                chunk_cb(bytes_piped)
    out_fh.flush()
    return bytes_piped

def decode_binary(blob: Union[str, bytes], encoding="utf-8") -> str:
    # Converts a binary type into a text type using given encoding.
    return blob if isinstance(blob, str) else blob.decode(encoding=encoding)

def load_binary_file(
    fname: Union[str, os.PathLike],
    *,
    read_cb: Optional[Callable[[int], None]] = None,
    quiet: bool = False,
) -> bytes:
    LOG.debug("Reading from %s (quiet=%s)", fname, quiet)
    with io.BytesIO() as ofh:
        try:
            with open(fname, "rb") as ifh:
                pipe_in_out(ifh, ofh, chunk_cb=read_cb)
        except FileNotFoundError:
            if not quiet:
                raise
        contents = ofh.getvalue()
    LOG.debug("Read %s bytes from %s", len(contents), fname)
    return contents


def load_text_file(
    fname: Union[str, os.PathLike],
    *,      
    read_cb: Optional[Callable[[int], None]] = None,
    quiet: bool = False,
) -> str:   
    return decode_binary(load_binary_file(fname, read_cb=read_cb, quiet=quiet))
```

`io.BytesIO()` [is documented as raising only exceptions which inherit from OSError](https://docs.python.org/3/library/io.html#high-level-module-interface) (with the exception of `ValueError` when used incorrectly, which we don't)

`OSError` [is really broad](https://docs.python.org/3/library/exceptions.html#exception-hierarchy) and would get thrown by open/read/close operations when file doesn't exist or reads are blocking or permissions error or anything like that. 

Since `load_binary_file()` returns `bytes`, `decode_binary()` should always call `.decode()` on the input (the only source of the aforementioned `UnicodeDecodeError`).

I don't see any other sources of exceptions except for programming errors which I don't think we want to silence (see [this PR for rational](https://github.com/canonical/cloud-init/pull/5202)).

Reply:
Thanks for checking and adding the exception!